### DISCUSSION TITLE
Need help,cannot close the popup (widget)

### DISCUSSION THREAD

**Discussion by Gelsendorf** [selenide/selenide#1912] [Q&A] [answered]
Need help,cannot close the popup (widget)

https://www.screenpresso.com/=aFcgc

**Comment by sbritprovectus** [comment]
Hello there.
If this dialog exists in frame please switch to the frame.
If not please try to click on this element using JS. 
If doesn't help try to send ESC using actions.

**Reply by Gelsendorf** [reply]
I thought about it, but I went through the keys that could close it, and no key closes it, except for clicking on the text itself, but this window is closed

**Comment by asolntsev** [comment]
The question is not related to Selenide at all. 
We don't see the entire html - probably the button is inside of a frame or shadow dom.

**Reply by Gelsendorf** [reply]
I am sorry, but could you help me to decide it. The button is shadow dom. How can I get around this, is there any command in selenium to close this html widget.Thanks. I can provide some additional information, really need to solve this

**Reply by Gelsendorf** [reply]
https://www.screenpresso.com/=oaE1d

**Reply by Gelsendorf** [reply]
Maybe need to use it $(shadowCss("#anyButton", "#shadow-host")).click();

**Reply by Gelsendorf** [reply]
something doesn't work, maybe I'm prescribing the wrong locators
![2022-08-07_14h01_09](https://user-images.githubusercontent.com/72403586/183287609-c4a5bd07-cb91-4854-bdad-4b9840d43dfe.png)

**Comment by Gelsendorf** [comment, marked as answer]
Done.Thanks
